.github/workflows/reporting-server-migration.yml

```diff
@@ -0,0 +1,59 @@
+name: Reporting Server Migration tests
+on:
+  pull_request:
+    paths:
+      - 'packages/reporting-server/**'
+  push:
+    branches:
+      - main
+jobs:
+  # Label of the container job
+  migrations:
+    # Containers must run in Linux based operating systems
+    runs-on: ubuntu-latest
+    # Service containers to run with `container-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres:latest
+        ports:
+          - 5432:5432
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      # Downloads a copy of the code in your repository before running CI tests
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Install python dependencies
+        run: pip3 install tortoise-orm pydantic aerich asyncpg 
+      - name: Copy migrations 
+        run: cp -r packages/reporting-server/migrations/ packages/reporting-server/test_migration/migrations/
+
+      - name: Connect to PostgreSQL and run migrations
+        # Runs a script that creates a PostgreSQL table, populates
+        # the table with data, and then retrieves the data.
+        run: cd packages/reporting-server/test_migration/ && aerich upgrade && aerich downgrade --yes -v 0
```

Comment:
After this test, the app will not work because I'm downgrading all the migrations with this command `aerich downgrade --yes -v 0`. My idea is to have 3 different workflows for this:
* The unit test running on the `reporting-server.yml` file
* `reporting-server-migration.yml` to test if I can upgrade all the migrations from scratch and if I can downgrade them without conflics. This will help us in case of a rollback.
* (In the future) ` reporting-server-e2e` where we will run all the current migrations, run the server and the UI to test the reports. I think this is covering your case.

Reply:
Does the downgrade command require both versions of the schema? For example I have a v1 schema and I want to upgrade to v2, aerich generates the .sql file for both v1 and v2. Now if I want to downgrade, do I need the v2 schema? Is it possible for aerich to perform the downgrade with just the v1 schema?

Asking this because I'm wondering how a rollback would work in practice. Suppose we push a bad version to production, the easiest way to rollback is to revert the commit, but doing so will lose the v2 schema.

Reply:
Aerich creates a table on the database where stores all the tables and the migration order. So you need to downgrade them in the correct order.

Answering your questions:
1. You will need V2 to go to V1.
2. You will have to downgrade in two steps. Apply downgrade on the database first and then revert the commit.

Reply:
Thanks, so this makes rolling back more complicated.